tidy3d/components/tcad/data/sim_data.py

```diff
@@ -151,6 +156,14 @@ def plot_mesh(
         matplotlib.axes._subplots.Axes
             The supplied or created matplotlib axes.
         """
+        if transpose:
+            # Please remove this warning once someone has verified that `transpose=True` works.
+            warn_untested_argument(
+                cls_name=type(self).__name__,
+                func_name="plot_mesh",
+                arg="transpose",
+                val="True",
+            )
```

Comment:
I did not find any tidy3d-notebook examples which invoke this function.

Of all the functions I modified for this PR, this is the function I have the most concerns about.  It needs to be tested.